app/controllers/oauth/applications_controller.rb

```diff
@@ -109,12 +121,49 @@ def set_user
     end
 
     def app_params
-      params.require(:doorkeeper_application).permit(
-        :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
-        :email_from_address, :confidential,
-        :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
-        :can_skip_oauth_screen,
-      )
+      if @user.is_administrator?
+        params.require(:doorkeeper_application).permit(
+          :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
+          :email_from_address, :confidential,
+          :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
+          :can_skip_oauth_screen, :member_ids
+        )
+      elsif @application.owner.member_ids.include?(@user.id)
+        params.require(:doorkeeper_application).permit(
+          :redirect_uri
+        )
+      end
+    end
+
+    def handle_member_ids(member_ids)
+      if valid_member_ids(member_ids)
+        if !member_ids.nil?
+          str_array = member_ids.split(" ")
+          mi_array = str_array.map(&:to_i)
+          #add the owner back to the member_ids array
+          mi_array.unshift(@application.owner.owner_ids[0])
+          @application.owner.member_ids = mi_array if @application.owner.member_ids != mi_array
+        end
+        return true
+      else
+        @application.errors.add(:owner, 'Member Ids must be a space separated list of integers')
+        return false
+      end
+    end
+
+    def valid_member_ids(member_ids)
+      if member_ids.nil? || member_ids.empty?
```

Comment:
FYI, in Rails, the ActiveSupport gem (which is always included) gives us the `blank?` method on basic classes: `nil.blank?` and `[].blank?` are both `true` so this line could be `if member_ids.blank?`

Reply:
I tried using .blank and .present and did not get the correct results which is why I used this line. It is removed now.